packages/components/pie-modal/src/modal.scss

```diff
@@ -232,29 +232,28 @@
     }
 
     &[isfooterpinned] .c-modal-content,
-    & .c-modal-scrollContainer {
+    & > .c-modal-scrollContainer {
         overflow-y: auto;
     }
 
-    & .c-modal-scrollContainer {
+    & > .c-modal-scrollContainer {
+        display: flex;
+        flex-direction: column;
+        height: 100%;
+
         // These are the shadows used to indicate scrolling above and below content
-        --bg-scroll-start: linear-gradient(var(--dt-color-container-default) 30%, rgba(255, 255, 255, 0));
         --bg-scroll-end: linear-gradient(rgba(255, 255, 255, 0), var(--dt-color-container-default) 70%) 0 100%;
-        --bg-scroll-top: radial-gradient(farthest-side at 50% 0, rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0));
-        --bg-scroll-bottom: radial-gradient(farthest-side at 50% 100%, rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0)) 0 100%;
+        --bg-scroll-bottom: radial-gradient(farthest-corner at 50% 100%, rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0)) 0 100%;
 
         // Sizes of the scroll shadows
-        --bg-size-scroll-start: 100% 40px;
         --bg-size-scroll-end: 100% 40px;
-        --bg-size-scroll-top: 100% 16px;
-        --bg-size-scroll-bottom: 100% 16px;
+        --bg-size-scroll-bottom: 100% 8px;
 
-        background: var(--bg-scroll-start), var(--bg-scroll-end), var(--bg-scroll-top), var(--bg-scroll-bottom);
+        background: var(--bg-scroll-end), var(--bg-scroll-bottom);
```

Comment:
QQ – is it expected that the shadow at the top of the modal doesn't appear now?

Reply:
yep, it isn't part of the design specs.